---
title: Life Sciences Datasets from the UCI Machine Learning Repository
author: Nathan Brouwer
date: '2020-12-20'
slug: life-sciences-datasets-from-the-uci-machine-learning-repository
categories:
  - ML
tags:
  - life sciences
math: no
meta: yes
toc: yes
---


<div id="TOC">
yes
</div>

<!--more-->
<p>The <a href="https://archive.ics.uci.edu/ml/index.php">UC Irvine Machine Learning Repository</a> is a collection of over 500 datasets related to machine learning (ML). There are a large number of entries related to the life sciences, including cancer medicine, bioinformatics, toxicology, and landcover classification. I’ve organized some of them below. I have not looked too deeply into anyone data set and so my classifications are all very preliminary.</p>
<div id="bioinformatics" class="section level2">
<h2>Bioinformatics</h2>
<p>There are many applications of ML in genomics and bioinformatics. While they often use standard ML techniques such as Support Vector Machines (SVL) or Neural Networks, the unique nature of DNA and protein sequence data requires specialized application of the methods. I’ve therefore tried to divided the datasets into those that could be examine used general data analysis and ML techniques and software, and those that require more specialized analysis.</p>
<div id="bioinformatics---general" class="section level3">
<h3>Bioinformatics - General</h3>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Codon+usage"><strong>Codon usage Data Set</strong></a></p>
<p><strong>URL:</strong> <a href="https://archive.ics.uci.edu/ml/datasets/Codon+usage" class="uri">https://archive.ics.uci.edu/ml/datasets/Codon+usage</a></p>
<p><strong>Abstract:</strong> “DNA codon usage frequencies of a large sample of diverse biological organisms from different taxa”</p>
<p>Khomtchouk BB, Liang Y, Nonner WF: ‘Codon usage bias levels predict taxonomic identity and genetic composition’. bioRxiv, 2020.</p>
<p>Nakamura Y, Gojobori T, Ikemura T: ‘Codon usage tabulated from international DNA sequence databases: status for the year 2000’. Nucleic Acids Research, 2000, 28:292.</p>
<p>Column 1: Kingdom
Column 2: DNAtype
Column 3: SpeciesID
Column 4: Ncodons
Column 5: SpeciesName
Columns 6-69: codon (header: nucleotide bases; entries: frequency of usage (5 digit floating point number))</p>
<p>"The ‘Kingdom’ is a 3-letter code corresponding to `xxx’ in the CUTG database name: ‘arc’(archaea), ‘bct’(bacteria), ‘phg’(bacteriophage), ‘plm’ (plasmid), ‘pln’ (plant), ‘inv’ (invertebrate), ‘vrt’ (vertebrate), ‘mam’ (mammal), ‘rod’ (rodent), ‘pri’ (primate), and ‘vrl’(virus) sequence entries. Note that the CUTG database does not contain ‘arc’ and ‘plm’ (these have been manually curated ourselves).</p>
<p>"The ‘DNAtype’ is denoted as an integer for the genomic composition in the species: 0-genomic, 1-mitochondrial, 2-chloroplast, 3-cyanelle, 4-plastid, 5-nucleomorph, 6-secondary_endosymbiont, 7-chromoplast, 8-leucoplast, 9-NA, 10-proplastid, 11-apicoplast, and 12-kinetoplast.</p>
<p>"The species identifier (‘SpeciesID’) is an integer, which uniquely indicates the entries of an organism. It is an accession identifier for each different species in the original CUTG database, followed by the first item listed in each genome.</p>
<p>"The number of codons (`Ncodons’) is the algebraic sum of the numbers listed for the different codons in an entry of CUTG. Codon frequencies are normalized to the total codon count, hence the number of occurrences divided by ‘Ncodons’ is the codon frequencies listed in the data file.</p>
<p>"The species’ name (‘SpeciesName’) is represented in strings purged of <code>comma' (which are now replaced by</code>space’). This is a descriptive label of the name of the species for data interpretations.</p>
<p>“Lastly, the codon frequencies (‘codon’) including ‘UUU’, ‘UUA’, ‘UUG’, ‘CUU’, etc., are recorded as floats (with decimals in 5 digits).”</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Yeast"><strong>Yeast Data Set</strong></a>
<strong>URL</strong>: <a href="https://archive.ics.uci.edu/ml/datasets/Yeast" class="uri">https://archive.ics.uci.edu/ml/datasets/Yeast</a>
<strong>Abstract</strong>: “Predicting the Cellular Localization Sites of Proteins”</p>
<p>“Expert Sytem for Predicting Protein Localization Sites in Gram-Negative Bacteria”, Kenta Nakai &amp; Minoru Kanehisa, PROTEINS: Structure, Function, and Genetics 11:95-110, 1991.</p>
<p>“A Knowledge Base for Predicting Protein Localization Sites in Eukaryotic Cells”, Kenta Nakai &amp; Minoru Kanehisa, Genomics 14:897-911, 1992.</p>
<p>Horton &amp; Nakai. 1996. “A Probablistic Classification System for Predicting the Cellular Localization Sites of Proteins”.Intelligent Systems in Molecular Biology, 109-115. St. Louis, USA 1996.</p>
<p><strong>Attribute Information:</strong>
Predicted Attribute: Localization site of protein. ( non-numeric ).
MIT NUC CYT ME1 ME2 ME3 VAC EXC</p>
<ol style="list-style-type: decimal">
<li>Sequence Name: Accession number for the SWISS-PROT database</li>
<li>mcg: McGeoch’s method for signal sequence recognition.</li>
<li>gvh: von Heijne’s method for signal sequence recognition.</li>
<li>alm: Score of the ALOM membrane spanning region prediction program.</li>
<li>mit: Score of discriminant analysis of the amino acid content of the N-terminal region (20 residues long) of mitochondrial and non-mitochondrial proteins.</li>
<li>erl: Presence of “HDEL” substring (thought to act as a signal for retention in the endoplasmic reticulum lumen). Binary attribute.</li>
<li>pox: Peroxisomal targeting signal in the C-terminus.</li>
<li>vac: Score of discriminant analysis of the amino acid content of vacuolar and extracellular proteins.</li>
<li>nuc: Score of discriminant analysis of nuclear localization signals of nuclear and non-nuclear proteins.</li>
</ol>
</div>
<div id="bioinformatics---potentially-general" class="section level3">
<h3>Bioinformatics - Potentially General</h3>
<p><a href="https://archive.ics.uci.edu/ml/datasets/M.+Tuberculosis+Genes">M. Tuberculosis Genes Data Set</a></p>
<p><strong>Abstract</strong>: "Data giving characteristics of each ORF (potential gene) in the *M. tuberculosis() bacterium. Sequence, homology (similarity to other genes) and structural information, and function (if known) are provided"</p>
<p>“The data is in Datalog format. Missing values are not explicit, but some genes have more relationships than others.”</p>
<p>King, R. and Karwath, A. and Clare, A. and Dehaspe, L. (2000). Accurate prediction of protein functional class in the M. tuberculosis and E. coli genomes using data mining, Comparative and Functional Genomics, 17, pp 283–293.</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Protein+Data">Protein Data Data Set</a>
<a href="https://archive.ics.uci.edu/ml/datasets/Protein+Data" class="uri">https://archive.ics.uci.edu/ml/datasets/Protein+Data</a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Molecular+Biology+%28Promoter+Gene+Sequences%29">Molecular Biology (Promoter Gene Sequences) Data Set</a></p>
<p><strong>Abstract</strong>: “E. Coli promoter gene sequences (DNA) with partial domain theory”</p>
<p>Harley, C. and Reynolds, R. 1987. “Analysis of E. Coli Promoter Sequences.” Nucleic Acids Research, 15:2343-2361.</p>
<p>Towell, G., Shavlik, J. and Noordewier, M. 1990. “Refinement of Approximate Domain Theories by Knowledge-Based Artificial Neural Networks.” In Proceedings of the Eighth National Conference on Artificial Intelligence (AAAI-90).</p>
<p>Mukund Deshpande and George Karypis. Evaluation of Techniques for Classifying Biological Sequences. PAKDD. 2002. [View Context].</p>
<p>Jinyan Li and Limsoon Wong. Using Rules to Analyse Bio-medical Data: A Comparison between C4.5 and PCL. WAIM. 2003. [View Context].</p>
<p>Aik Choon Tan and David Gilbert. An Empirical Comparison of Supervised Machine Learning Techniques in Bioinformatics. APBC. 2003. [View Context].</p>
<p><strong>Attribute Information:</strong></p>
<ol style="list-style-type: decimal">
<li>One of {+/-}, indicating the class (“+” = promoter).</li>
<li>The instance name (non-promoters named by position in the 1500-long nucleotide sequence provided by T. Record).
3-59. The remaining 57 fields are the sequence, starting at position -50 (p-50) and ending at position +7 (p7). Each of these fields is filled by one of {a, g, t, c}.</li>
</ol>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Molecular+Biology+%28Splice-junction+Gene+Sequences%29">Molecular Biology (Splice-junction Gene Sequences)</a></p>
<p><strong>Abstract</strong>: “Primate splice-junction gene sequences (DNA) with associated imperfect domain theory”</p>
<p><strong>Problem Description:</strong> “Splice junctions are points on a DNA sequence at which ‘superfluous’ DNA is removed during the process of protein creation in higher organisms. The problem posed in this dataset is to recognize, given a sequence of DNA, the boundaries between exons (the parts of the DNA sequence retained after splicing) and introns (the parts of the DNA sequence that are spliced out). This problem consists of two subtasks: recognizing exon/intron boundaries (referred to as EI sites), and recognizing intron/exon boundaries (IE sites). (In the biological community, IE borders are referred to a ‘acceptors’ while EI borders are referred to as ‘donors’.)”</p>
<p>“This dataset has been developed to help evaluate a”hybrid" learning algorithm (KBANN) that uses examples to inductively refine preexisting knowledge."</p>
<p><strong>Attribute Information:</strong></p>
<ol style="list-style-type: decimal">
<li>One of {n ei ie}, indicating the class.</li>
<li>The instance name.
3-62. The remaining 60 fields are the sequence, starting at position -30 and ending at position +30. Each of these fields is almost always filled by one of {a, g, t, c}. Other characters indicate ambiguity among the standard characters…</li>
</ol>
<p>M. O. Noordewier and G. G. Towell and J. W. Shavlik, 1991; “Training Knowledge-Based Neural Networks to Recognize Genes in DNA Sequences”. Advances in Neural Information Processing Systems, volume 3, Morgan Kaufmann.
[Web Link]</p>
<p>Jinyan Li and Limsoon Wong. Using Rules to Analyse Bio-medical Data: A Comparison between C4.5 and PCL. WAIM. 2003.</p>
<p>Xiaojin Zhu. Label Propagation for Eukaryotic Splice Junction Identification. 2002. [View Context].</p>
<p>Mukund Deshpande and George Karypis. Evaluation of Techniques for Classifying Biological Sequences. PAKDD. 2002.</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/PubChem+Bioassay+Data"><strong>PubChem Bioassay Data Data Set</strong></a>
<strong>URL</strong>: <a href="https://archive.ics.uci.edu/ml/datasets/PubChem+Bioassay+Data" class="uri">https://archive.ics.uci.edu/ml/datasets/PubChem+Bioassay+Data</a></p>
<p><strong>Abstract</strong>: “These highly imbalanced bioassay datasets are from the differing types of screening that can be performed using HTS technology. 21 datasets were created from 12 bioassays.”</p>
<p>Chen, Wang &amp; Zhang: 2011. The use of classification trees for bioinformatics. DOI: 10.1002/widm.14</p>
<p>Chin Yee Liew, Xiao Hua Ma and Chun Wei Yap. Consensus model for identification of novel PI3K inhibitors in large chemical library. Journal of Computer-Aided Molecular Design. Volume 24, Number 2, 131-141, DOI: 10.1007/s10822-010-9321-0</p>
<p>Genetic Algorithm-Neural Network (GANN): a study of neural network activation functions and depth of genetic algorithm search applied to feature selection.</p>
<p>Dong Ling Tong and Robert Mintram International Journal of Machine Learning and Cybernetics. Volume 1, Numbers 1-4, 75-87, DOI: 10.1007/s13042-010-0004-x</p>
</div>
<div id="bioinformatics---specialized" class="section level3">
<h3>Bioinformatics - specialized</h3>
<p><a href="https://archive.ics.uci.edu/ml/datasets/p53+Mutants"><strong>p53 Mutants Data Set</strong></a>
<a href="https://archive.ics.uci.edu/ml/datasets/p53+Mutants" class="uri">https://archive.ics.uci.edu/ml/datasets/p53+Mutants</a></p>
<p>Danziger et al. (2009) Predicting Positive p53 Cancer Rescue Regions Using Most Informative Positive (MIP) Active Learning, PLOS Computational Biology, 5(9), e1000498</p>
<p>Danziger et al. (2007) Choosing where to look next in a mutation sequence space: Active Learning of informative p53 cancer rescue mutants, Bioinformatics, 23(13), 104-114.</p>
<p>Danziger et al. (2006) Functional census of mutation sequence spaces: the example of p53 cancer rescue mutants, IEEE/ACM transactions on computational biology and bioinformatics / IEEE, ACM, 3, 114-125.</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Mice+Protein+Expression"><strong>Mice Protein Expression Data Set</strong></a>
<strong>URL</strong>: <a href="https://archive.ics.uci.edu/ml/datasets/Mice+Protein+Expression" class="uri">https://archive.ics.uci.edu/ml/datasets/Mice+Protein+Expression</a></p>
<p>Higuera C, Gardiner KJ, Cios KJ (2015) Self-Organizing Feature Maps Identify Proteins Critical to Learning in a Mouse Model of Down Syndrome. PLoS ONE 10(6): e0129126. [Web Link] journal.pone.0129126</p>
<p>Ahmed MM, Dhanasekaran AR, Block A, Tong S, Costa ACS, Stasko M, et al. (2015) Protein Dynamics Associated with Failed and Rescued Learning in the Ts65Dn Mouse Model of Down Syndrome. PLoS ONE 10(3): e0119491. [Web Link]</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/HIV-1+protease+cleavage()"><strong>HIV-1 protease cleavage Data Set</strong></a>
<strong>URL</strong>: <a href="https://archive.ics.uci.edu/ml/datasets/HIV-1+protease+cleavage" class="uri">https://archive.ics.uci.edu/ml/datasets/HIV-1+protease+cleavage</a></p>
<p>Thorsteinn Rögnvaldsson, Liwen You and Daniel Garwicz, ‘State of the art prediction of HIV-1 protease cleavage sites’ Bioinformatics (2015) 31 (8): 1204-1210. doi: 10.1093/bioinformatics/btu810,</p>
</div>
<div id="bioinformatics---rna-seq" class="section level3">
<h3>Bioinformatics - RNA-Seq</h3>
<p><a href="https://archive.ics.uci.edu/ml/datasets/gene+expression+cancer+RNA-Seq"><strong>gene expression cancer RNA-Seq Data Set</strong></a>
<a href="https://archive.ics.uci.edu/ml/datasets/gene+expression+cancer+RNA-Seq" class="uri">https://archive.ics.uci.edu/ml/datasets/gene+expression+cancer+RNA-Seq</a>
<strong>URL</strong>: <a href="https://archive.ics.uci.edu/ml/datasets/gene+expression+cancer+RNA-Seq" class="uri">https://archive.ics.uci.edu/ml/datasets/gene+expression+cancer+RNA-Seq</a>
<strong>Abstract:</strong> This collection of data is part of the RNA-Seq (HiSeq) PANCAN data set, it is a random extraction of gene expressions of patients having different types of tumor: BRCA, KIRC, COAD, LUAD and PRAD.</p>
<p>Weinstein, John N., et al. ‘The cancer genome atlas pan-cancer analysis project.’ Nature genetics 45.10 (2013): 1113-1120.</p>
</div>
<div id="bioinformatics---rna-seq-1" class="section level3">
<h3>Bioinformatics - RNA-Seq</h3>
<p><a href="https://archive.ics.uci.edu/ml/datasets/chipseq"><strong>chipseq Data Set</strong></a></p>
</div>
</div>
<div id="bioinformatics---protein-struycture-chemistry-function" class="section level2">
<h2>Bioinformatics - Protein Struycture, chemistry &amp; function</h2>
<p>[<strong>Physicochemical Properties of Protein Tertiary Structure Data Set</strong>]
(<a href="https://archive.ics.uci.edu/ml/datasets/Physicochemical+Properties+of+Protein+Tertiary+Structure" class="uri">https://archive.ics.uci.edu/ml/datasets/Physicochemical+Properties+of+Protein+Tertiary+Structure</a>)</p>
<p>RMSD-Size of the residue.
F1 - Total surface area.
F2 - Non polar exposed area.
F3 - Fractional area of exposed non polar residue.
F4 - Fractional area of exposed non polar part of residue.
F5 - Molecular mass weighted exposed area.
F6 - Average deviation from standard exposed area of residue.
F7 - Euclidian distance.
F8 - Secondary structure penalty.
F9 - Spacial Distribution constraints (N,K Value).</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/QSAR+androgen+receptor">QSAR androgen receptor Data Set</a>
<strong>URL</strong>: <a href="https://archive.ics.uci.edu/ml/datasets/QSAR+androgen+receptor" class="uri">https://archive.ics.uci.edu/ml/datasets/QSAR+androgen+receptor</a></p>
</div>
<div id="medicine" class="section level2">
<h2>Medicine</h2>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Bone+marrow+transplant%3A+children"><strong>Bone marrow transplant: children Data Set</strong></a></p>
<p>WrÃ³bel, Å, GudyÅ›, A, Sikora, M (2017) Learning rule sets from survival data,
BMC Bioinformatics, 18(1):285</p>
<p>KaÅ‚wak, K, Porwolik, J, Mielcarek, M et al. (2010) Higher CD34+ and CD3+ cell doses in the graft promote long-term survival, and have no impact on the incidence of severe acute or chronic graft-versus-host disease after in vivo t cell-depleted unrelated donor hematopoietic stem cell transplantation in children, Biology of Blood and Marrow Transplantation, 16(10): 1388-1401</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Caesarian+Section+Classification+Dataset"><strong>Caesarian Section Classification Dataset Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Acute+Inflammations"><strong>Acute Inflammations Data Set</strong></a>
<a href="https://archive.ics.uci.edu/ml/datasets/Acute+Inflammations" class="uri">https://archive.ics.uci.edu/ml/datasets/Acute+Inflammations</a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Vertebral+Column"><strong>Vertebral Column Data Set</strong></a>
<a href="https://archive.ics.uci.edu/ml/datasets/Vertebral+Column" class="uri">https://archive.ics.uci.edu/ml/datasets/Vertebral+Column</a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Fertility"><strong>Fertility Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Heart+failure+clinical+records"><strong>Heart failure clinical records Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Parkinson%27s+Disease+Classification"><strong>Parkinson’s Disease Classification Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Autistic+Spectrum+Disorder+Screening+Data+for+Children++"><strong>Autistic Spectrum Disorder Screening Data for Children Data Set</strong></a></p>
</div>
<div id="cancer" class="section level2">
<h2>Cancer</h2>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Gastrointestinal+Lesions+in+Regular+Colonoscopy"><strong>Gastrointestinal Lesions in Regular Colonoscopy Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Cervical+cancer+%28Risk+Factors%29"><strong>Cervical cancer (Risk Factors) Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/HCC+Survival"><strong>HCC Survival Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Coimbra"><strong>Breast Cancer Coimbra Data Set</strong></a></p>
</div>
<div id="anthropology" class="section level2">
<h2>Anthropology</h2>
<p><a href="https://archive.ics.uci.edu/ml/datasets/StoneFlakes"><strong>StoneFlakes Data Set</strong></a></p>
</div>
<div id="landcover-classification" class="section level2">
<h2>Landcover classification</h2>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Forest+type+mapping"><strong>Forest type mapping Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Amphibians"><strong>Amphibians Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Urban+Land+Cover"><strong>Urban Land Cover Data Set</strong></a></p>
</div>
<div id="ecology" class="section level2">
<h2>Ecology</h2>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Abalone"><strong>Abalone Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Forest+Fires"><strong>Forest Fires Data Set</strong></a>
<strong>Abstract:</strong> “This is a difficult regression task, where the aim is to predict the burned area of forest fires, in the northeast region of Portugal, by using meteorological and other data.”</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Anuran+Calls+%28MFCCs%29"><strong>Anuran Calls (MFCCs) Data Set</strong></a></p>
</div>
<div id="toxicology" class="section level2">
<h2>Toxicology</h2>
<p><a href="https://archive.ics.uci.edu/ml/datasets/QSAR+aquatic+toxicity"><strong>QSAR aquatic toxicity Data Set</strong></a></p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/QSAR+fish+toxicity"><strong>QSAR fish toxicity Data Set</strong></a></p>
<p>M. Cassotti, D. Ballabio, R. Todeschini, V. Consonni. A similarity-based QSAR model for predicting acute toxicity towards the fathead minnow (Pimephales promelas), SAR and QSAR in Environmental Research (2015), 26, 217-243; doi: 10.1080/1062936X.2015.1018938</p>
</div>
